src/Infrastructure/Export/ExcelExport.cs

```diff
@@ -0,0 +1,38 @@
+﻿using ClosedXML.Excel;
+using FSH.WebApi.Application.Common.Exporters;
+using System.ComponentModel;
+using System.Data;
+
+namespace FSH.WebApi.Infrastructure.Export;
+
+public class ExcelExport : IExporter
+{
+    public DataTable Convert<T>(IList<T> data)
+    {
+        PropertyDescriptorCollection properties = TypeDescriptor.GetProperties(typeof(T));
+        DataTable table = new DataTable("table", "table");
+        foreach (PropertyDescriptor prop in properties)
+            table.Columns.Add(prop.Name, Nullable.GetUnderlyingType(prop.PropertyType) ?? prop.PropertyType);
+        foreach (T item in data)
+        {
+            DataRow row = table.NewRow();
+            foreach (PropertyDescriptor prop in properties)
+                row[prop.Name] = prop.GetValue(item) ?? DBNull.Value;
+            table.Rows.Add(row);
+        }
+        return table;
+    }
+
+    public MemoryStream ExportToAsync(DataTable dt)
+    {
+        using (XLWorkbook wb = new XLWorkbook())
+        {
+            wb.Worksheets.Add(dt);
+            using (MemoryStream stream = new MemoryStream())
+            {
+                wb.SaveAs(stream);
+                return stream;
```

Comment:
As this is returned inside a using block, this will return a disposed stream.
So no need for the using here (the caller is responsible for closing the stream, which will actually happen by aspnetcore after the stream is returned to the client (using the File ActionResult).

Also you can return just Stream from this method... Doesn't need to be a specific memorystream